### ISSUE TITLE
JSpecifyMode: function returning a @Nullable is incorrectly handled

### ISSUE THREAD

**Issue by nvervelle** [uber/NullAway#1671]
Hello, it's probably related to #1532 , but as I'm not entirely sure, I prefer to open a dedicated issue (feel free to close it if it's the same underlying problem).

I've tried activating JSpecifyMode in my project and I encounter the following false positive

```java
    return webClient
        .post()
        .uri(properties.accessUrl())
        .contentType(MediaType.APPLICATION_FORM_URLENCODED)
        .body(BodyInserters.fromFormData(formData))
        .retrieve()
        .toEntity(JsonNode.class)
        .mapNotNull(ResponseEntity::getBody)
        .mapNotNull(this::updateAccessToken)
        .switchIfEmpty(Mono.error(() -> new IllegalStateException("Unable to get access token")));
```

The error message is

```
error: [NullAway] referenced method returns @Nullable, but functional interface method java.util.function.Function.apply(T) returns @NonNull
          .mapNotNull(ResponseEntity::getBody)
```

but `ResponseEntity.getBody()` (from `org.springframework.http.ResponseEntity`) is defined as

```java
  public @Nullable T getBody() {
    return this.body;
  }
```

and `Mono.mapNotNull()` is defined as

```java
  public final <R> Mono<R> mapNotNull(Function <? super T, ? extends @Nullable R> mapper) {
    ...
  }

```

**Comment by msridhar** [comment]
Hi @nvervelle can you point me at your project so I can try it, or some reproducer?  This is probably related to wildcards, and we've done a lot of work to improve support for them, but it's off by default right now.  I'd like to see if that support fixes this one.

**Comment by nvervelle** [comment]
Hi @msridhar , thanks for the quick answer.

Sorry, the project is not publicly available, it's for my company. Is there an option to enable the improvement for wildcards support on my side to try it directly ?

Otherwise, I can try producing a reproducible small example, but not immediately.

**Comment by nvervelle** [comment]
Hi @msridhar .

In fact, it was easy

```java
package testjspecify;

import org.springframework.http.ResponseEntity;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;
import tools.jackson.databind.JsonNode;

public class TestJSpecify1671 {

  public static Mono<String> testJSpecify() {
    return WebClient.create()
        .post()
        .uri("https://example.com")
        .retrieve()
        .toEntity(JsonNode.class)
        .mapNotNull(ResponseEntity::getBody)
        .mapNotNull(jsonNode -> jsonNode.get("access_token").asString())
        .switchIfEmpty(Mono.error(() -> new IllegalStateException("Unable to get access token")));
  }
}
```

Error:

```
.../TestJSpecify1671.java:16: error: [NullAway] referenced method returns @Nullable, but functional interface method java.util.function.Function.apply(T) returns @NonNull
        .mapNotNull(ResponseEntity::getBody)
                    ^

```

Versions:
* spring 7.0.8
* reactor 3.8.6
* jackson 3.1.4

**Comment by msridhar** [comment]
I've confirmed this issue is fixed on the latest master branch with our wildcard support enabled.  I'll work on getting this support out into a release soon.  I still don't recommend enabling the wildcard support with the current release.

**Comment by nvervelle** [comment]
Hi @msridhar .

If I understand correctly, you say it works only if enabling the wildcard support, but you don't recommend enabling it, so I can't activate JSpecifyMode for the moment ?

**Comment by msridhar** [comment]
@nvervelle that comment was for the previous release.  The support is shipped in 0.14.0 under the JSpecifyExperimental flag.  We have a few open bugs already to improve JSpecifyExperimental.  I suggest waiting until release 0.14.1 and then trying the JSpecifyExperimental flag.  Sorry for being unclear.
